src/Features/CSharp/Portable/UpgradeProject/CSharpUpgradeProjectCodeFixProvider.cs

```diff
@@ -61,6 +52,11 @@ private static LanguageVersion RequiredVersion(ImmutableArray<Diagnostic> diagno
                 {
                     max = max > required ? max : required;
                 }
+                else if (diagnostic.Id == "CS8652")
```

Comment:
Is there any reason CS8652 doesn't set DiagnosticPropertyConstants.RequiredLanguageVersion?

Reply:
(I wouldn't block this PR, but still a good question. Not sure what other code might exist that's grabbing that property and expecting uniformity.)

Reply:
Special versions like "preview" aren't, as you noted before, well-ordered in the version list. That means that fixers (like the upgrade project fixer) may produce bad results if they're naively looking at that value and doing simple integer comparison.

Opting-in seems safer, since no one could have possibly seen this diagnostic before.